semgrep-core/src/mlgrep/cli-lib/Semgrep_CLI_lib.ml

```diff
@@ -0,0 +1,79 @@
+(*
+   Library defining the semgrep command-line interface.
+
+   This module determines the subcommand invoked on the command line
+   and has another module handle it as if it were an independent command.
+   We don't use Cmdliner to dispatch subcommands because it's a too
+   complicated and we never show a help page for the whole command anyway
+   since we fall back to the 'scan' subcommand if none is given.
+*)
+
+open Printf
+
+(* This is used to determine if we should fall back to assuming 'scan'. *)
+let known_subcommands =
+  [ "ci"; "login"; "logout"; "lsp"; "publish"; "scan"; "shouldafound" ]
+
+(* Exit with a code that a proper semgrep implementation would never return.
+   Uncaught OCaml exception result in exit code 2.
+   This is to ensure that the tests that expect error status 2 fail. *)
+let missing_subcommand () =
+  eprintf "This semgrep subcommand is not implemented\n%!";
+  exit Error.not_implemented_in_mlgrep
```

Comment:
same, we do not need the exit call here.

Reply:
thx